pyccel/parser/semantic.py

```diff
@@ -2699,6 +2699,12 @@ def _handle_is_operator(self, IsClass, expr, **settings):
         var1 = self._visit(expr.lhs)
         var2 = self._visit(expr.rhs)
 
+        if (var1 == var2):
```

Comment:
@saidctb Do you know how `==` is implemented for instances of `Variables`? Is `==` sufficient or should we use `is` here?